cognite_toolkit/_cdf_tk/loaders/_resource_loaders/function_loaders.py

```diff
@@ -163,6 +165,41 @@ def _create_hash_values(cls, function_rootdir: Path) -> str:
                 hash_value += f";{new_entry}"
         return hash_value
 
+    def get_function_required_capabilities(
+        self, items: Sequence[FunctionWrite] | None, read_only: bool
+    ) -> list[Capability] | list[Capability]:
+        """
+        Get required capabilities for working with CDF Functions and their associated files.
+
+        This method determines the necessary permissions needed for function operations,
+        with dataset-scoped file access when possible. When functions are associated with
+        specific datasets, it will request file permissions only for those datasets
+        rather than requiring all-scope file access.
+        """
+        if not items and items is not None:
```

Comment:
Nitpick:

`if items is not None and not items:` applies the same logic, but it's more readable.